specification/dartLangSpec.tex

```diff
@@ -19062,8 +19062,8 @@ \section*{Appendix: Integer Implementations}
 The \code{int} type represents integers.
 The specification is written with 64-bit two's complement integers as the
 intended implementation, but when Dart is compiled to JavaScript,
-the implementation of \code{int} will instead use the JavaScript
-number type.
+the implementation of \code{int} will instead use the JavaScript number type,
+with bit operations applied to the low 32 bits as an unsigned value.
```

Comment:
> with bit-wise operations only work on the first 32 bits of an integer value. 
> The operands are first truncated to 32-bit two's complement integers, then 32-bit bitwise operations are applied, and finally the resulting 32 bits are interpreted as an unsigned 32-bit integer.

I don't want to say that the operands are unsigned, because `>>` is sign-extending (so it must have a sign, so it can't be "unsigned"). The above description is precise. JavaScript converts bit-operation operands to 32-bit integers using their spec-function ToInt32, then it performs the bit-operation on that, and finally Dart performs a `>>>0` on the result, which calls `ToUint32` with the result.

Reply:
The point is that "when compiling to JS, Dart uses the JavaScript number type" seems to imply that the semantics of all operations are taken directly from similarly named operations in JS. So I wanted to add "but with respect to bit operations, the semantics is different".

I changed the wording to make this more explicit.